doc/auth.md

```diff
@@ -0,0 +1,320 @@
+# Kuadrant Auth
+
+A Kuadrant AuthPolicy custom resource:
+
+1. Allows it to target Gateway API networking resources such as [HTTPRoutes](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.HTTPRoute) and [Gateways](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.Gateway), using these resources to obtain additional context, i.e., which traffic workload (HTTP attributes, hostnames, user attributes, etc) to enforce auth.
+2. Allows to specify which specific subsets of the targeted network resource to apply the auth rules to.
+3. Abstracts the details of the underlying external authorization protocol and configuration resources, that have a much broader remit and surface area.
+4. Supports cluster operators to set overrides (soon) and defaults that govern what can be done at the lower levels.
```

Comment:
perhaps this should be reflected somewhere else? Wondering if we could have an issue label for AuthPolicy for example and link to the GH issues instead?

Reply:
We have the same line in the RL docs: https://github.com/Kuadrant/kuadrant-operator/blob/ee2a4b36132ed597aeb47ff5eef354267a867451/doc/rate-limiting.md?plain=1#L8. This is basically our promise to support inherited policy attachment eventually.

Reply:
So I think it would still be good to call out that this supports cluster operators setting a default policy based on the Gateway hierarchy? Yes its not defaults overrides but is is a form of default WDYT?

Reply:
Changed a bit. Please let me know what you think.

Reply:
+1